src/slq.jl

```diff
@@ -12,65 +12,65 @@ end
 Generating a random vector based on Rademacher Distribution
 """
 function randomRademacherVector(x::Int64)
-    mp = [-1 1];
-    v = rand(1,x);
-
-    for i = range(1, length = x)
-        v[i] = mp[(v[i] < 0.5) + 1];
-    end
-
-    return v
+    return rand(-1.0:2.0:1.0, x)
```

Comment:
Allow passing a number type `T` and return a random vector with that type as the `eltype`. For example:
```julia
o = one(T)
t = 2*o
rand(-o:t:o, x)
```
This enables this algorithm to generalize to `Float32`, higher precision types, etc.